src/lib.rs

```diff
@@ -160,6 +160,14 @@ pub const SECONDS_PER_YEAR: f64 = 31_557_600.0;
 pub const SECONDS_PER_TROPICAL_YEAR: f64 = 31_556_925.974_7;
 /// `SECONDS_PER_SIDERAL_YEAR` corresponds to the number of seconds per sideral year from [NIST](https://www.nist.gov/pml/special-publication-811/nist-guide-si-appendix-b-conversion-factors/nist-guide-si-appendix-b9#TIME).
 pub const SECONDS_PER_SIDERAL_YEAR: f64 = 31_558_150.0;
+/// `SECONDS_GPS_TAI_OFFSET` is the number of seconds from the TAI epoch to the
+/// GPS epoch (UTC midnight of January 5th 1980; cf.
+/// https://gssc.esa.int/navipedia/index.php/Time_References_in_GNSS#GPS_Time_.28GPST.29)
+pub const SECONDS_GPS_TAI_OFFSET: f64 = SECONDS_PER_YEAR * 80.0 + SECONDS_PER_DAY * 4.0;
```

Comment:
This PR looks great, thanks! Just one question: I'm not sure I follow the math on the `SECONDS_GPS_TAI_OFFSET`. How does 4 days + 80 seconds corresponds to the January 5th 1980 UTC ? Note that I never deal with GPS epoch so I most likely am misunderstanding something crucial here and if I were to bet, I'd bet your math is correct.

From the reference:
> It starts at 0h UTC (midnight) of January 5th to 6th 1980 (6.d0). At that epoch, the difference TAI−UTC was 19 seconds.

Does that quote imply the offset should be 4 days and 19 seconds instead of 4 days and 80 seconds?

Reply:
Thanks for this comment - it made me scrutinise things more closely. It is confusing, but the correct value is:

```rust
pub const SECONDS_GPS_TAI_OFFSET: f64 = 80.0 * SECONDS_PER_YEAR + 4.0 * SECONDS_PER_DAY + 19.0;
````
(same as before but +19s).  I think you missed the 80 being applied to seconds per *year*, so this is 80 years, 4 days and 19 seconds.  I was banging my head for a while thinking that this takes us to midnight of January 5th 1980, but I verified that this is right with some new tests:

```rust
    let gps_epoch = Epoch::from_tai_seconds(SECONDS_GPS_TAI_OFFSET);
    assert_eq!(
        gps_epoch.as_gregorian_str(TimeSystem::UTC),
        "1980-01-06T00:00:00 UTC"
    );
    assert_eq!(
        gps_epoch.as_gregorian_str(TimeSystem::TAI),
        "1980-01-06T00:00:19 TAI"
    );
```

Hopefully this is foolproof!